source/cppexpose/include/cppexpose/variant/Variant.inl

```diff
@@ -80,7 +80,7 @@ T Variant::value(const T & defaultValue) const
     // Type of variant is the wanted type
     if (m_value && typeid(T) == m_value->type())
     {
-        return reinterpret_cast<DirectValue<T> *>(m_value)->value();
```

Comment:
Here we use a ```reinterpret_cast``` whereas later we use ```static_cast``` in similar construct. Which one is preferred?